internal/buildinformation/build_info.go

```diff
@@ -0,0 +1,83 @@
+package buildinformation
+
+import (
+	"runtime/debug"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+)
+
+const (
+	goarch      string = "GOARCH"
+	goos        string = "GOOS"
+	vcstype     string = "vcs"
+	vcsrevision string = "vcs.revision"
+	vcstime     string = "vcs.time"
+)
+
+type BuildInfo struct {
+	commitSha       string
+	architecture    string
+	vcsTime         string
+	vcsType         string
+	operatingSystem string
+	goVersion       string
+}
+
+// GetBuildInfo returns the current Build Information as a BuildInfo type (struct) using a built-in debug package and its ReadBuildInfo() function.
```

Comment:
No need to indicate the type being returned nor the details of it being a type nor implementation details.
An alternative documentation might be:
// GetBuildInfo returns build related information of the binary or an error